packages/gestalt/src/Masonry/defaultTwoColumnModuleLayout.js

```diff
@@ -283,20 +294,21 @@ const defaultTwoColumnModuleLayout = <T: { +[string]: mixed }>({
     };
 
     if (hasTwoColumnItems) {
-      // If the number of items to position is greater that the batch size
-      // we identify the batch with the two column item and apply the graph only to those items
       // Currently we only support one two column item at the same time, more items will be supporped soon
       const twoColumnIndex = itemsWithoutPositions.indexOf(twoColumnItems[0]);
 
+      // Skip the graph logic if the two column item batch is on the first line
+      const skipGraph =
+        heights.every((height) => height === 0) &&
+        itemsWithoutPositions.length <= TWO_COL_ITEMS_MEASURE_BATCH_SIZE;
+
       // If the number of items to position is greater that the batch size
       // we identify the batch with the two column item and apply the graph only to those items
-      const shouldBatchItems = itemsWithoutPositions.length > TWO_COL_ITEMS_MEASURE_BATCH_SIZE;
-      const splitIndex =
-        twoColumnIndex + TWO_COL_ITEMS_MEASURE_BATCH_SIZE > itemsWithoutPositions.length
-          ? itemsWithoutPositions.length - TWO_COL_ITEMS_MEASURE_BATCH_SIZE
-          : twoColumnIndex;
-      const pre = shouldBatchItems ? itemsWithoutPositions.slice(0, splitIndex) : [];
-      const batchWithTwoColumnItems = shouldBatchItems
+      const splitIndex = skipGraph
+        ? twoColumnIndex
+        : calculateSplitIndex(itemsWithoutPositions.length, twoColumnIndex);
+      const pre = splitIndex ? itemsWithoutPositions.slice(0, splitIndex) : [];
```

Comment:
if you change above to return 0, this can also just be `itemsWithoutPositions.slice(0, splitIndex)` since split(0,0) is the same as []